internal/controller/atlasproject/team_reconciler.go

```diff
@@ -58,14 +53,8 @@ func (r *AtlasProjectReconciler) teamReconcile(
 			return result.ReconcileResult(), nil
 		}
 
-		atlasClient, orgID, err := r.AtlasProvider.SdkClientSet(teamCtx.Context, connectionSecretKey, log)
-		if err != nil {
-			result := workflow.Terminate(workflow.AtlasAPIAccessNotConfigured, err)
-			setCondition(teamCtx, api.ReadyType, result)
-			return result.ReconcileResult(), nil
-		}
-		teamCtx.OrgID = orgID
-		teamCtx.SdkClient = atlasClient.SdkClient20231115008
+		teamCtx.OrgID = workflowCtx.OrgID
+		teamCtx.SdkClient = workflowCtx.SdkClientSet.SdkClient20231115008
```

Comment:
You're passing nil here because the client was set to a different field inn the project controller:
https://github.com/mongodb/mongodb-atlas-kubernetes/pull/2158/files#diff-0d383863ebbfca7afb47ff12d141447ce823fbcd48870db9722d7b5a53cb3e15R165

This is making the Gov (because of teams) and Teams tests to fail

```suggestion
		teamCtx.SdkClient = workflowCtx.SdkClientSet.SdkClient
```